### PR TITLE
HPM configuration

### PR DESCRIPTION
**Pull request by cetvrtak**
Fronter configuration for enabling HPM.

Actual option to be disabled for users after the review.

### CODE REVIEW

File: EU4toV2/Source/Configuration.cpp [hunk 1]
```diff
@@ -41,6 +41,9 @@ void Configuration::instantiate(std::istream& theStream, bool (*DoesFolderExist)
 		verifyVic2DocumentsPath(Vic2DocumentsPath, DoesFolderExist);
 		LOG(LogLevel::Info) << "Vic2 documents path: " << Vic2DocumentsPath;
 	});
+	registerKeyword("targetGameModPath", [this, DoesFolderExist](const std::string& unused, std::istream& theStream) {
```
Comment (Idhrendur) [inline EU4toV2/Source/Configuration.cpp:44]: No need to capture DoesFolderExist

File: EU4toV2/Source/Configuration.cpp [hunk 2]
```diff
@@ -109,6 +117,11 @@ void Configuration::instantiate(std::istream& theStream, bool (*DoesFolderExist)
 	parseStream(theStream);
 	clearRegisteredKeywords();
 	setOutputName();
+	if (isHpmEnabled())
+	{
+		Vic2Path = Vic2ModPath + "/HPM";
```
Comment (Idhrendur) [inline EU4toV2/Source/Configuration.cpp:122]: You need tests for this behavior.

File: EU4toV2/Source/Configuration.cpp [hunk 3]
```diff
@@ -109,6 +117,11 @@ void Configuration::instantiate(std::istream& theStream, bool (*DoesFolderExist)
...
+		Vic2DocumentsPath += "/HPM";
```
Comment (Idhrendur) [inline EU4toV2/Source/Configuration.cpp:123]: Vic2DocumentsPath isn't used outside of verifying the folder exists and tests, so you don't need to update it. And even if you did, I don't think adding HPM to it is what you'd want.

File: EU4toV2/Source/Configuration.cpp (hunk 2 above)
Comment (cetvrtak) [reply EU4toV2/Source/Configuration.cpp:122]: Would it suffice to have two tests: if it is enabled `C:\Vic2ModPath\HPM`, else `C:\Vic2Path`?

**Review by Idhrendur — APPROVED** [review]

File: EU4toV2/Source/Configuration.cpp [hunk 4]
```diff
@@ -41,6 +41,10 @@ void Configuration::instantiate(std::istream& theStream, bool (*DoesFolderExist)
 		verifyVic2DocumentsPath(Vic2DocumentsPath, DoesFolderExist);
 		LOG(LogLevel::Info) << "Vic2 documents path: " << Vic2DocumentsPath;
 	});
+	registerKeyword("targetGameModPath", [this](const std::string& unused, std::istream& theStream) {
```
Comment (Zemurin) [inline EU4toV2/Source/Configuration.cpp:44]: "hybridModSource" - we're creating a hybrid mod using this one as source.

File: EU4toV2/Source/Configuration.cpp [hunk 5]
```diff
@@ -41,6 +41,10 @@ void Configuration::instantiate(std::istream& theStream, bool (*DoesFolderExist)
...
+		Vic2ModPath = commonItems::getString(theStream);
```
Comment (Zemurin) [inline EU4toV2/Source/Configuration.cpp:45]: hybridModSourcePath

File: EU4toV2/Source/Configuration.cpp [hunk 6]
```diff
@@ -41,6 +41,10 @@ void Configuration::instantiate(std::istream& theStream, bool (*DoesFolderExist)
...
+		LOG(LogLevel::Info) << "Vic2 mod path: " << Vic2ModPath;
```
Comment (Zemurin) [inline EU4toV2/Source/Configuration.cpp:46]: "Merging with mod at: "

File: EU4toV2/Source/Configuration.cpp [hunk 7]
```diff
@@ -99,6 +103,11 @@ void Configuration::instantiate(std::istream& theStream, bool (*DoesFolderExist)
 		convertAll = convertAllString == "yes";
 		LOG(LogLevel::Info) << "Convert All: " << convertAllString;
 	});
+	registerKeyword("hpm_enabled", [this](std::istream& theStream) {
```
Comment (Zemurin) [inline EU4toV2/Source/Configuration.cpp:106]: hybrid_enabled

File: EU4toV2/Source/Configuration.cpp [hunk 8]
```diff
@@ -99,6 +103,11 @@ void Configuration::instantiate(std::istream& theStream, bool (*DoesFolderExist)
...
+		const auto hpmEnabledString = commonItems::getString(theStream);
+		vic2Mod = VIC2MOD(std::stoi(hpmEnabledString));
```
Comment (Zemurin) [inline EU4toV2/Source/Configuration.cpp:108]: hybridMod = HYBRIDMOD(...)

File: EU4toV2/Source/Configuration.h [hunk 9]
```diff
@@ -38,6 +38,7 @@ class Configuration: commonItems::parser
 		enum class AFRICARESET { ResetAfrica = 1, LeaveAfrica = 2 };
 		enum class ABSORBCOLONIES { AbsorbNone = 1, AbsorbSome = 2, AbsorbAll = 3 };
 		enum class LIBERTYDESIRE { Loyal = 1, Disloyal = 2, Rebellious = 3 };
+		enum class VIC2MOD { None = 1, HPM = 2 };
```
Comment (Zemurin) [inline EU4toV2/Source/Configuration.h:41]: HYBRIDMOD

File: EU4toV2/Source/Configuration.h [hunk 10]
```diff
@@ -52,6 +53,7 @@ class Configuration: commonItems::parser
 		[[nodiscard]] auto getRandomiseRgos() const { return randomiseRgos; }
 		[[nodiscard]] auto getConvertAll() const { return convertAll; }
 		[[nodiscard]] auto getAfricaReset() const { return africaReset; }
+		[[nodiscard]] bool isHpmEnabled() const { return vic2Mod == VIC2MOD::HPM; }
```
Comment (Zemurin) [inline EU4toV2/Source/Configuration.h:56]: This for example is fine.

**Review by Zemurin — CHANGES_REQUESTED** [review]

File: EU4toV2/Source/Configuration.cpp (hunk 4 above)
Comment (cetvrtak) [reply EU4toV2/Source/Configuration.cpp:44]: This was recycled from another converter. It points to where Victoria II mods are stored.

Comment (cetvrtak) [reply EU4toV2/Source/Configuration.cpp:44]: Though, I'm 99% certain that it was only needed with the "any mod" mind frame. So I'm removing it entirely.

Comment (cetvrtak) [reply EU4toV2/Source/Configuration.cpp:44]: Nope, I was wrong. I need it to tell the Converter that HPM folder is inside the `C:\Vic2ModPath`, unless we can assume that the path is always at `C:\Vic2Path\mod` which is a bit risky. As someone once said people do all kinds of crazy things.

**Review by Zemurin — APPROVED** [review]

**Pull request merged**